.github/workflows/format.yml

```diff
@@ -0,0 +1,21 @@
+name: format check
+
+on:
+  pull_request
+
+jobs:
+  format:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: install mise
+        uses: jdx/mise-action@v2
+      - name: run formatter
+        run: mise run fmt
+      - name: check for changes
+        run: |
+          if [[ -n $(git status --porcelain) ]]; then
+            echo "code formatting issues found. please run 'mise run fmt' locally and commit the changes."
```

Comment:
Nitpick: let's use sentence formatting for things like names and documentation/prompts.

Reply:
I hate sentence formatting, but you are the boss

fixed